### PR TITLE
high_level: Accept lists for data arguments in train, score and predict

### LINKED ISSUE #1222: high level: train: Accept list as well as dict
Use model.config.features to build dict

### PR DESCRIPTION
**Pull request by mHash1m**
fixes #1222 
Signed-off-by: Hashim Chaudry <hashimchaudry23@gmail.com>

### CODE REVIEW

File: dffml/high_level/ml.py [hunk 1]
```diff
@@ -51,6 +51,19 @@ async def train(model, *args: Union[BaseSource, Record, Dict[str, Any]]):
     >>>
     >>> asyncio.run(main())
     """
+    if any(isinstance(arg, list) for arg in args):
+        if isinstance(model.config.predict, Features):
```
Comment (pdxjohnny) [inline dffml/high_level/ml.py:55]:
```suggestion
        if hasattr(model.config, "predict") and isinstance(model.config.predict, Features):
```

File: dffml/high_level/ml.py [hunk 2]
```diff
@@ -51,6 +51,19 @@ async def train(model, *args: Union[BaseSource, Record, Dict[str, Any]]):
...
+            )
+        else:
+            args = list_records_to_dict(
+                [feature.name for feature in model.config.features]
+                + [model.config.predict.name],
+                *args,
+            )
     async with contextlib.AsyncExitStack() as astack:
         # Open sources
         sctx = await astack.enter_async_context(records_to_sources(*args))
```
Comment (pdxjohnny) [inline dffml/high_level/ml.py:69]: Let's make model a keyword argument and if it's present, then we can support converting lists to Record instances. We should raise an exception if we were unable to convert a list to a Record.

```suggestion
        sctx = await astack.enter_async_context(records_to_sources(*args, model=model))
```

File: dffml/high_level/ml.py [hunk 3]
```diff
@@ -51,6 +51,19 @@ async def train(model, *args: Union[BaseSource, Record, Dict[str, Any]]):
     >>>
     >>> asyncio.run(main())
     """
+    if any(isinstance(arg, list) for arg in args):
```
Comment (pdxjohnny) [inline dffml/high_level/ml.py:54]:
```suggestion
    if hasattr(model.config, "features") and any(isinstance(arg, list) for arg in args):
```

**Review by pdxjohnny — CHANGES_REQUESTED** [review]

File: dffml/high_level/ml.py (hunk 2 above)
Comment (mHash1m) [reply dffml/high_level/ml.py:69]: Assuming this was supposed to be for the function `list_records_to_dict()`

**Pull request merged**